packages/cache/types/service.ts

```diff
@@ -0,0 +1,47 @@
+import { z } from "zod";
+import { logger } from "@formbricks/logger";
+import type { CacheError, Result } from "./error";
+import { ErrorCode, err, ok } from "./error";
+
+// Validation schemas for cache service
+export const ZCacheKey = z
```

Comment:
This can also include the brand since zod supports this as well.


```suggestion
export const ZCacheKey = z
  .string()
  .min(1, "Cache key cannot be empty")
  .refine((key) => key.trim().length > 0, "Cache key cannot be empty or whitespace only")
  .brand("CacheKey");

export type CacheKey = z.infer<typeof ZCacheKey>;
```

the type inferred from this works fine across the current uses of the `CacheKey` type and we also dont need to make a separate type for this.
If this is done, I think we can move this type to `types/key.ts`

Reply:
Great suggestion!